ironfish-cli/src/commands/wallet/multisig/sign.ts

```diff
@@ -289,19 +411,45 @@ export class SignMultisigTransactionCommand extends IronfishCommand {
 
   private async performAggregateCommitments(
     client: RpcClient,
+    multisigClient: MultisigTcpClient | null,
     accountName: string,
     commitment: string,
     identities: string[],
+    totalParticipants: number,
     unsignedTransaction: UnsignedTransaction,
   ) {
-    this.log(
-      `Enter ${identities.length - 1} commitments of the participants (excluding your own)`,
-    )
+    let commitments: string[] = []
+    if (!multisigClient) {
+      this.log(
+        `Enter ${identities.length - 1} commitments of the participants (excluding your own)`,
+      )
 
-    const commitments = await ui.collectStrings('Commitment', identities.length - 1, {
-      additionalStrings: [commitment],
-      errorOnDuplicate: true,
-    })
+      commitments = await ui.collectStrings('Commitment', identities.length - 1, {
+        additionalStrings: [commitment],
+        errorOnDuplicate: true,
+      })
+    } else {
+      multisigClient.submitSigningCommitment(commitment)
+
+      multisigClient.onSigningStatus.on((message) => {
```

Comment:
Why is there multiple ways of getting the signing commitments?

Reply:
I agree it's simpler for the client if the server broadcasts the full list once it's complete instead of broadcasting them one at a time. The only thing we'd lose there is the ability to show progress (e.g., '3/4 commitments')

This polls the status in addition to listening for broadcast commitments in case you miss a broadcast by being behind or disconnected.

Reply:
Once we add encryption, this code will need to decrypt and verify that it's a valid commitment so the server won't actually know once all valid commitments are there. It works now because it's plain text.